eb_hooks.py

```diff
@@ -744,6 +744,26 @@ def pre_configure_hook_LAMMPS_zen4(self, *args, **kwargs):
     else:
         raise EasyBuildError("LAMMPS-specific hook triggered for non-LAMMPS easyconfig?!")
 
+def pre_configure_hook_nvhpc_x86_64(self, *args, **kwargs):
+    """
+    pre-configure hook for nvhpc
+    - search and replace operations in the ec dict
+    """
+    if self.name == "NVHPC":
```

Comment:
This is good for now, but there is quite a bit of discussion right now about changing the naming in the EasyBuild context. We are currently naming the _compilers only_ `NVHPC` but we should perhaps be defining a toolchain hierarchy for `NVHPC` since they also contain MPI and some math libraries. This may lead to the compilers being called something like `nvidia_compilers` or `NVHPC` becoming a fatter toolchain.

Reply:
You're also going to need some versioning clauses here for things you have actually tested.

This is definitely a corner case at present as EESSI itself will not ship his toolchain (currently), and I wonder if we should not allow an environment variable to force the use of the hook, something like
```suggestion
    if self.name == "NVHPC":
        force_nvhpc_hook = 'EESSI_FORCE_NVHPC_HOOK'
        if self.version in [] or os.getenv(force_nvhpc_hook, False):
            ...
        else:
            print_msg(f"Not using exisiting hook for {self.name}/{self.version}, if you wish to force this please set the envvar {force_nvhpc_hook}")
```